src/DynamoCoreWpf/ViewModels/Core/ConnectorPinViewModel.cs

```diff
@@ -0,0 +1,252 @@
+﻿using System;
+using System.Linq;
+using Dynamo.Configuration;
+using Dynamo.Graph;
+using Dynamo.Selection;
+using Dynamo.UI.Commands;
+using Dynamo.Wpf.ViewModels.Core;
+using Newtonsoft.Json;
+
+namespace Dynamo.ViewModels
+{
+    public partial class ConnectorPinViewModel : ViewModelBase
+    {
+        #region Events
+        public event EventHandler RequestSelect;
+        public virtual void OnRequestSelect(Object sender, EventArgs e)
+        {
+            if (RequestSelect != null)
+            {
+                RequestSelect(this, e);
+            }
+        }
+
+        public event EventHandler RequestRedraw;
+        public virtual void OnRequestRedraw(Object sender, EventArgs e)
+        {
+            if (RequestRedraw != null)
+            {
+                RequestRedraw(this, e);
+            }
+        }
+
+        public event EventHandler RequestRemove;
+        public virtual void OnRequestRemove(Object sender, EventArgs e)
+        {
+            RequestRemove(this, e);
+        }
+
+        #endregion
+
+        #region Properties
+
+        private ConnectorPinModel _model;
+
+        [JsonIgnore]
+        public readonly WorkspaceViewModel WorkspaceViewModel;
+        private int zIndex = Configurations.NodeStartZIndex; // initialize the start Z-Index of a note to the same as that of a node
+        internal static int StaticZIndex = Configurations.NodeStartZIndex;
```

Comment:
why does this need to be static?

Reply:
It follows the same pattern of the notes/nodes. I believe it sets the baseline ZIndex for views of this type once and should be immutable..

Reply:
hmm okay, still we have these two why not just have one of the?
```csharp
private int zIndex = Configurations.NodeStartZIndex; // initialize the start Z-Index of a note to the same as that of a node
internal static int StaticZIndex = Configurations.NodeStartZIndex;

Reply:
also the comment next to `zIndex` is weird

Reply:
we will also need some comments on this as we would need to explain why we are using the Nodes zindex in the pin

Reply:
I guess that one is the baseline.. Where as the non-static one varies based on what you are doing with the mouse. ie if you are hovering over it, it temporarily gets bumped up.